asmcomp/mach.ml

```diff
@@ -156,9 +156,17 @@ let rec instr_iter f i =
       | _ ->
           instr_iter f i.next
 
+let operation_is_pure = function
+  | Icall_ind | Icall_imm _ | Itailcall_ind | Itailcall_imm _
+  | Iextcall _ | Istackoffset _ | Istore _ | Ialloc _
+  | Iintop(Icheckbound) | Iintop_imm(Icheckbound, _) | Iopaque -> false
+  | Ispecific sop -> Arch.operation_is_pure sop
+  | _ -> true
+
 let operation_can_raise op =
```

Comment:
It's unchanged by this patch, but I find it strange that tail calls are not included here. Is this correct?

Reply:
All dataflow analyses in the back-end have special cases for tail calls, since their control-flow is not that of a normal operation.  So, `is_pure` and `can_raise` should never be called on a tailcall instruction.  I'd rather not change this as part of this PR.

Reply:
Fair enough!